lambdas/DumpSnippets/lambda_function.py

```diff
@@ -0,0 +1,77 @@
+import os
+import json
+import boto3
+import zipfile
+import base64
+
+# Initialization
+s3 = boto3.client('s3', 'us-west-2')
+try:
+    bn = os.environ['BucketName']
+except:
+    print 'Must specify "BucketName" env var!'
+    raise error
+try:
+    zb = os.environ['ZipsBucket']
+except KeyError:
+    zb = 'codesplain-zips'
+snippets_bucket = boto3.resource('s3').Bucket(bn)
+zips_bucket     = boto3.resource('s3').Bucket(zb)
+
+
+def zip_snippets(snippet_keys):
+    """Writes the contents of the snippets with given keys to a zip file,
+       returns the path to the zip."""
+    zip_path = '/tmp/dump.zip'
+    with zipfile.ZipFile(zip_path, 'w') as dump_zip:
```

Comment:
The user doesn't care, she wants her zip faster! We're being charged by time anyhow, it behooves us to make it faster.